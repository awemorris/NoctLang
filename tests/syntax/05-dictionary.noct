func main() {
    // Dictionary literal
    a = {
        "aaa": "bbb",
        ccc: "ddd"
    };
    print(a.aaa);
    print(a["ccc"]);

    // Dictionary assignment
    dict = {};
    dict["aaa"] = "value1";
    dict.bbb = 123;

    // Dictionary overwrite
    for (k, v in dict) {
    	print("k = " + k + ", v = " + v);
    }
    dict.aaa = "AFTER";
    for (k, v in dict) {
    	print("k = " + k + ", v = " + v);
    }
}
