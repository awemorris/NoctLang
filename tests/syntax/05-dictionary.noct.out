bbb
ddd
k = bbb, v = 123
k = aaa, v = value1
k = bbb, v = 123
k = aaa, v = AFTER
